From 0c8cac40a6ee69e106f3b034624890b98cad5b01 Mon Sep 17 00:00:00 2001
From: Paul Eggert <eggert@cs.ucla.edu>
Date: Mon, 12 Oct 2020 16:15:27 -0700
Subject: [PATCH] Put dummy pacificnew into rearguard tarball

Problem reported by Mike Cullinan in:
https://mm.icann.org/pipermail/tz/2020-October/029309.html
* Makefile (tzdata$(VERSION)-rearguard.tar.gz):
Put an empty pacificnew file into the rearguard tarball.
* NEWS: Mention this.
---
 Makefile | 6 +++++-
 NEWS     | 7 +++++++
 2 files changed, 12 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 28972c6..95f0a06 100644
--- a/Makefile
+++ b/Makefile
@@ -1022,10 +1022,14 @@ tzdata$(VERSION)-rearguard.tar.gz: rearguard.zi set-timestamps.out
 		done
 		sed '1s/$$/-rearguard/' \
 		  <version >tzdata$(VERSION)-rearguard.dir/version
+		: The dummy pacificnew pacifies TZUpdater 2.3.1 and earlier.
+		touch -md 2020-10-12T22:53:00Z \
+		  tzdata$(VERSION)-rearguard.dir/pacificnew
 		touch -cmr version tzdata$(VERSION)-rearguard.dir/version
 		LC_ALL=C && export LC_ALL && \
 		  (cd tzdata$(VERSION)-rearguard.dir && \
-		   tar $(TARFLAGS) -cf - $(COMMON) $(DATA) $(MISC) | \
+		   tar $(TARFLAGS) -cf - \
+			$(COMMON) $(DATA) $(MISC) pacificnew | \
 		     gzip $(GZIPFLAGS)) >$@.out
 		mv $@.out $@
 
diff --git a/NEWS b/NEWS
index 298ad4d..87d68d9 100644
--- a/NEWS
+++ b/NEWS
@@ -6,12 +6,19 @@ Unreleased, experimental changes
     Fiji starts DST later than usual, on 2020-12-20.
 
   Changes to future timestamps
+
     Fiji will start DST on 2020-12-20, instead of 2020-11-08 as
     previously predicted.  DST will still end on 2021-01-17.
     (Thanks to Raymond Kumar and Alan Mintz.)  Assume for now that
     the later-than-usual start date is a one-time departure from the
     recent pattern.
 
+  Changes to build procedure
+
+    Rearguard tarballs now contain an empty file pacificnew.
+    Some older downstream software expects this file to exist.
+    (Problem reported by Mike Cullinan.)
+
 
 Release 2020b - 2020-10-06 18:35:04 -0700
 
-- 
1.8.3.1

